homeassistant/components/habitica/sensor.py

```diff
@@ -39,6 +41,78 @@
     "class": SensorType("Class", "mdi:sword", None, ["stats", "class"]),
 }
 
+
+@dataclass(kw_only=True, frozen=True)
+class HabitipySensorEntityDescription(SensorEntityDescription):
+    """Habitipy Sensor Description."""
+
+    value_path: list[str]
+
+
+SENSOR_DESCRIPTIONS: dict[str, HabitipySensorEntityDescription] = {
+    "name": HabitipySensorEntityDescription(
+        key="name",
+        translation_key="name",
+        value_path=["profile", "name"],
+    ),
+    "hp": HabitipySensorEntityDescription(
+        key="hp",
+        translation_key="hp",
+        native_unit_of_measurement="HP",
+        suggested_display_precision=0,
+        value_path=["stats", "hp"],
+    ),
+    "maxHealth": HabitipySensorEntityDescription(
+        key="maxHealth",
+        translation_key="maxhealth",
+        native_unit_of_measurement="HP",
+        value_path=["stats", "maxHealth"],
+    ),
+    "mp": HabitipySensorEntityDescription(
+        key="mp",
+        translation_key="mp",
+        native_unit_of_measurement="MP",
+        suggested_display_precision=0,
+        value_path=["stats", "mp"],
+    ),
+    "maxMP": HabitipySensorEntityDescription(
+        key="maxMP",
+        translation_key="maxmp",
+        native_unit_of_measurement="MP",
+        value_path=["stats", "maxMP"],
+    ),
+    "exp": HabitipySensorEntityDescription(
+        key="exp",
+        translation_key="exp",
+        native_unit_of_measurement="XP",
+        value_path=["stats", "exp"],
+    ),
+    "toNextLevel": HabitipySensorEntityDescription(
+        key="toNextLevel",
+        translation_key="tonextlevel",
+        native_unit_of_measurement="XP",
+        value_path=["stats", "toNextLevel"],
+    ),
+    "lvl": HabitipySensorEntityDescription(
+        key="lvl",
+        translation_key="lvl",
+        native_unit_of_measurement="Lvl",
+        value_path=["stats", "lvl"],
+    ),
+    "gp": HabitipySensorEntityDescription(
+        key="gp",
+        translation_key="gp",
+        native_unit_of_measurement="🜚",  # alchemy symbol for gold
```

Comment:
I mean, the fact that we need to understand this with a comments means that users also won't get it (since we can't show this)

Reply:
Thought it might be a nice detail, as it is a fantasy themed game. The symbol was added to Unicode in 2010 and shows correctly for me, also on my android phone, but if it might be a problem on some devices i'd consider GP as unit

Reply:
I'd just say "gold"

Reply:
can units be translated? Otherwise, GP seems also to be the term used by Habitica https://habitica.fandom.com/wiki/Gold_Points

Reply:
Also works, there's an architectural proposal to add translatable UoMs which is accepted, so maybe in the future